ANTLR/src/Compiler/ContextualAnalysis/TypeCheckerVisitor.java

```diff
@@ -189,7 +197,12 @@ public Integer visitFuncdef(FuncdefContext ctx) {
         // Check that the types correspond to each other.
         for (int i = 0; i < stmtsTypes.size(); i++) {
             Integer someStmtType = stmtsTypes.get(i);
-            checkReturnTypeCorrespondence(someStmtType, ctx);
+            try {
+                checkReturnTypeCorrespondence(someStmtType, ctx);
+            } catch (Exception e){
+                errorListener.ThrowError(e.getMessage(), stmts.get(i).stmt(), ctx.type().start);
+            }
+
         }
```

Comment:
Kan vi rykke loopet ind i try-catch i statementet i stedet?

Reply:
Kan egentlig godt lide at man kan se lige præcis hvad der burde throwe. Ellers risikerer vi at fange de exceptions som stmtsTypes.get(i) kunne finde på at kaste, og det vil vi ikke. Hvad synes du om det?

Reply:
3. mulighed: Hvorfor ikke bare throwe erroren på linje 224 i stedet, hvor der står følgende?

`throw new Exception("Does not return expected type in function definition: " + functionId.toString());`